### PR TITLE
Asyn support

### LINKED ISSUE #10: Implement asynchronous execution of commands
Currently asynchronous execution is not straight away supported by erlimem

### PR DESCRIPTION
**Pull request by shamis**
Fixes #10

### CODE REVIEW

File: src/erlimem_session.erl [hunk 1]
```diff
@@ -344,23 +344,27 @@ handle_info({_Ref,{StmtRef,Result}}, #state{stmts=Stmts}=State) when is_pid(Stmt
...
-                {noreply, State#state{stmts=Stmts}};
-            Term ->
-                ?Debug("Async __RX__ ~p For ~p", [Term, From]),
-                gen_server:reply(From, Term),
-                {noreply, State}
+        {error, Exception} ->
+            ?Debug("to ~p throw~n~p~n", [From, Exception]),
+            gen_server:reply(From,  {error, Exception}),
+            {noreply, State};
+        {ok, #stmtResult{stmtRef  = StmtRef} = SRslt} ->
```
Comment (acautin) [inline src/erlimem_session.erl]: Are not all the calls wrapped now ?

**Comment by shamis** [comment]
@c-bik can [this case block](https://github.com/K2InformaticsGmbH/erlimem/pull/15/files#diff-b52cbfeb3c7e9608390b4b9f049f6850R349) be replaced as its already handled [here ](https://github.com/K2InformaticsGmbH/erlimem/pull/15/files#diff-b52cbfeb3c7e9608390b4b9f049f6850R360).

File: src/erlimem_session.erl [hunk 2]
```diff
@@ -344,23 +344,27 @@ handle_info({_Ref,{StmtRef,Result}}, #state{stmts=Stmts}=State) when is_pid(Stmt
...
+            gen_server:reply(From, Rslt),
+            {noreply, State#state{stmts=Stmts}};
+        {reply, Term} ->
+            ?Debug("Sync __RX__ ~p For ~p", [Term, From]),
+            gen_server:reply(From, Term),
+            {noreply, State};
+        Resp ->
+            ?Debug("Async __RX__ ~p For ~p", [Resp, From]),
+            P ! Resp,
+            {noreply, State}
```
Comment (c-bik) [inline src/erlimem_session.erl]: refactor:
```erlang
handle_info({Form, {reply, Resp}}, State) ->
    case Resp of
        {error, Exception} -> ...
        {ok, #stmtResult{...}} ->
            ...
            ?Debug("statement ~p stored in ~p", [StmtRef, [S|| {S,_} <- State#state.stmts]]),
        ...
    end,
    {noreply, State};
handle_info({{P,_}, Resp}, State) ->
    ?Debug("Async __RX__ ~p For ~p", [Resp, P]),
    P ! Resp,
    {noreply, State};
```

Comment (acautin) [inline src/erlimem_session.erl]: But still we don't need the error clause unless we are doing something special with it, if we are just sending it back, we could just leave it to the last clause.

**Pull request merged**